mlflow/server/js/src/experiment-tracking/pages/prompts/components/ModelConfigForm.tsx

```diff
@@ -0,0 +1,197 @@
+import { FormUI, RHFControlledComponents, Typography, useDesignSystemTheme } from '@databricks/design-system';
+import { useFormContext } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+interface ModelConfigFormProps {
+  namePrefix?: string; // For nested forms, e.g., "modelConfig."
+}
+
+export const ModelConfigForm = ({ namePrefix = '' }: ModelConfigFormProps) => {
+  const { theme } = useDesignSystemTheme();
+  const intl = useIntl();
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext();
+
+  const fieldName = (name: string) => `${namePrefix}${name}`;
+
+  const getError = (name: string) => {
+    const parts = fieldName(name).split('.');
+    let error: any = errors;
+    for (const part of parts) {
+      error = error?.[part];
+    }
+    return error;
+  };
+
+  return (
+    <div
+      css={{
+        border: `1px solid ${theme.colors.border}`,
+        borderRadius: theme.general.borderRadiusBase,
+        padding: theme.spacing.md,
+      }}
+    >
+      <Typography.Title level={4} css={{ marginBottom: theme.spacing.sm }}>
+        <FormattedMessage
+          defaultMessage="Model Configuration"
+          description="Section header for model configuration in prompt creation"
+        />
+      </Typography.Title>
+
+      <div css={{ display: 'flex', flexDirection: 'column', gap: theme.spacing.md }}>
+        <div>
+          <FormUI.Label htmlFor={fieldName('modelName')}>
+            <FormattedMessage defaultMessage="Model" description="Label for model name input in model config form" />
+          </FormUI.Label>
+          <RHFControlledComponents.Input
+            control={control}
+            id={fieldName('modelName')}
+            componentId={`mlflow.prompts.model_config.${fieldName('modelName')}`}
+            name={fieldName('modelName')}
```

Comment:
As we discussed, can we separate model name and provider name?

Reply:
yes, done!